sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileInputStream.java

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake;
+
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.common.StorageInputStream;
+import com.azure.storage.file.datalake.models.DataLakeRequestConditions;
+import com.azure.storage.file.datalake.models.DataLakeStorageException;
+import com.azure.storage.file.datalake.models.FileRange;
+import com.azure.storage.file.datalake.models.PathProperties;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Provides an input stream to read a given Data Lake file resource.
+ */
+public final class DataLakeFileInputStream extends StorageInputStream {
```

Comment:
I wonder if we should just wrap a BlobInputStream instead so we dont have to replicate some of this logic.

See blob-nio's implementation of BlobInputStream

Reply:
I dislike the idea of having to override everything `StorageInputStream` overrides and keeping up with that as we maintain that class. I'd rather just automatically get any changes there and have a smidge of duplicate logic as a result.

We could port most of dispatchRead() up to StorageInputStream and only override the absolute necessities? Future subclasses could still override all of dispatchRead() if needed.

Reply:
Makes sense